chore: upgrade to prisma 5.4.2

zomars opened the pull request:
## What does this PR do?

<!-- Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context. List any dependencies that are required for this change. -->

Upgrades prisma to 5.4.2

## Type of change

<!-- Please delete bullets that are not relevant. -->

- [x] Chore (refactoring code, technical debt, workflow improvements)

## How should this be tested?

- Checks should be passing

## Mandatory Tasks

- [x] Make sure you have self-reviewed the code. A decent size PR without self-review might be rejected.

zomars committed:
chore: upgrade to prisma 5.3.1

zomars commented:
Current dependencies on/for this PR:
* main
  * **PR #11666** <a href="https://app.graphite.dev/github/pr/calcom/cal.com/11666" target="_blank"><img src="https://static.graphite.dev/graphite-32x32-black.png" alt="Graphite" width="10px" height="10px"/></a>  👈

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/calcom/cal.com/11666?utm_source=stack-comment).

PeerRich reviewed (dismissed):
i dont believe i saw breaking changes from 5.2 to 5.3.1

PeerRich committed:
new yarn.lock

PeerRich committed:
yarn merge fix

Dhoni77 committed:
docs: add an alternate approach to seed local database (#11307)

Co-authored-by: Peer Richelsen <peer@cal.com>

crowdin-bot committed:
New Crowdin translations by Github Action

crowdin-bot committed:
New Crowdin translations by Github Action

crowdin-bot committed:
New Crowdin translations by Github Action

crowdin-bot committed:
New Crowdin translations by Github Action

anikdhabal committed:
chore: throw toast when adding a new date overwrite (#11764)

Co-authored-by: Peer Richelsen <peer@cal.com>
Co-authored-by: Peer Richelsen <peeroke@gmail.com>

crowdin-bot committed:
New Crowdin translations by Github Action

PeerRich committed:
feat: added dark mode to login, logout, password reset (#11765)

Co-authored-by: sean-brydon <55134778+sean-brydon@users.noreply.github.com>

devkiran committed:
fix: Hide SAML/OIDC login button if no sso connection exists (Self hosted instance) (#10903)

Co-authored-by: zomars <zomars@me.com>
Co-authored-by: Peer Richelsen <peeroke@gmail.com>
Co-authored-by: Syed Ali Shahbaz <52925846+alishaz-polymath@users.noreply.github.com>

alishaz-polymath committed:
add yarn lockfile section for contributors (#11769)

Co-authored-by: Peer Richelsen <peeroke@gmail.com>

jannesblobel committed:
fix: update inlang project settings file (#11442)

zomars committed:
fix: API build

hariombalhara committed:
feat: Embed - Introduce `prerender` instruction - Lightning fast popup experience (#11303)

Co-authored-by: Peer Richelsen <peeroke@gmail.com>
Co-authored-by: Omar López <zomars@me.com>

hariombalhara committed:
test: Add collective scheduling tests (#11670)

Udit-takkar committed:
fix: add prisma import (#11781)

Udit-takkar committed:
fix: subteam avatar flciker (#11773)

sean-brydon committed:
feat: overlay your calendar (#11693)

* Init header + login modal component

* Add calendar settings for authed user

* Local storage and using query params for toggle

* Toggle connect screen if query param present and no session

* Local storage + store + way more than that should be in single commit

* Display busy events on weekly view

* Confirm booking slot of overlap exists

* use chevron right when on column view

* Show hover card - overlapping date times

* Invalidate on switch

* FIx clearing local storage when you login to another account

* Force re-render on url state (atom quirks)

* Add loading screen

* Add dialog close

* Remove extra grid config

* Translations

* [WIP] - tests

* fix: google calendar busy times (#11696)

Co-authored-by: CarinaWolli <wollencarina@gmail.com>

* New Crowdin translations by Github Action

* fix: rescheduled value DB update on reschedule and insights view cancelleds (#11474)

* v3.3.5

* fix minutes string (#11703)

Co-authored-by: CarinaWolli <wollencarina@gmail.com>

* Regenerated yarn.lock

* Add error component + loader

* await tests

* disable tests - add note

* Refactor to include selected time

* use no-scrollbar

* Fix i18n

* Fix tablet toolbar

* overflow + i18n

* Export empty object as test is TODO

* Uses booker timezone

* Fix hiding switch too early

* Handle selected timezone

* Fix timezone issues

* Fix timezone issues

---------

Co-authored-by: Carina Wollendorfer <30310907+CarinaWolli@users.noreply.github.com>
Co-authored-by: CarinaWolli <wollencarina@gmail.com>
Co-authored-by: Crowdin Bot <support+bot@crowdin.com>
Co-authored-by: alannnc <alannnc@gmail.com>
Co-authored-by: Alex van Andel <me@alexvanandel.com>
Co-authored-by: Peer Richelsen <peer@cal.com>
Co-authored-by: Peer Richelsen <peeroke@gmail.com>

grzpab committed:
refactor: implementation of locale calculated server-side (#11534)

emrysal committed:
fix: Remove credentials and some ts fixes (#11794)

anikdhabal committed:
fix position and add transition (#11799)

DexterStorey committed:
hotfix for #11626 (#11807)

emrysal committed:
fix: Bring back dir attr using Intl (#11806)

Co-authored-by: Omar López <zomars@me.com>

samueldenzil committed:
fix-issue#11797 (#11800)

emrysal committed:
v3.3.7

grzpab committed:
chore: ensure compat type definitions do not break existing types (#11741)

emrysal committed:
fix: newLocale fallback to 'en' locale (#11808)

zomars committed:
hotfix: missing locale

Nit

Udit-takkar committed:
refactor: event type settings improvements (#11615)

Udit-takkar committed:
refactor: org settings (#11682)

gitstart-app[bot] committed:
Change approach (#11681)

Co-authored-by: gitstart-calcom <gitstart-calcom@users.noreply.github.com>
Co-authored-by: GitStart-Cal.com <121884634+gitstart-calcom@users.noreply.github.com>

sean-brydon committed:
test: api-middleware (#11825)

* tests/api-middleware

* Skip if true

* Remove prisma from response mock

grzpab committed:
hotfix: use session locale when dealing with SSG-ed pages (#11831)

emrysal committed:
fix: Revert deletion of getLocaleFromRequest in #11534 (#11835)

CarinaWolli committed:
fix: missing last day of month

## What does this PR do?

- Fix that the last day in Oktober was always shown as unavailable
This happened when availability was in Europe/London timezone

- Fix that the last day of November was always shown as unavailable
This happened also for other timezones

## Type of change

- [x] Bug fix (non-breaking change which fixes an issue)

## How should this be tested?

I recommend testing it in staging instead of local, as I experienced some different behaviors on local 

- Reproduce the 2 bugs as described before my fix
- If you were able to reproduce it, see if it is fixed with my fixes in this PR (last day of octber and November should be shown as available)
- Check if availability starts and ends at the correct time on days were DST happens

zomars committed:
upgrade prisma to 5.4.2

zomars committed:
Merge branch 'main' into 10-02-chore_upgrade_to_prisma_5.3.1

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/calcom/cal.com?pullRequest=11666) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you all sign our [Contributor License Agreement](https://cla-assistant.io/calcom/cal.com?pullRequest=11666) before we can accept your contribution.<br/>**8** out of **10** committers have signed the CLA.<br/><br/>:white_check_mark: Dhoni77<br/>:white_check_mark: devkiran<br/>:white_check_mark: jannesblobel<br/>:white_check_mark: anikdhabal<br/>:white_check_mark: DexterStorey<br/>:white_check_mark: emrysal<br/>:white_check_mark: grzpab<br/>:white_check_mark: Udit-takkar<br/>:x: crowdin-bot<br/>:x: samueldenzil<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/calcom/cal.com?pullRequest=11666) it.</sub>

zomars committed:
Update CalendarService.test.ts

millsp committed:
chore(prisma): undo skipped type checks (#11454)

zomars committed:
Update get.handler.ts

zomars commented:
Fixed booking filters which now doesn't allow undefined values in "AND" clauses in "where" filters.

zomars commented on packages/trpc/server/routers/viewer/bookings/get.handler.ts:
```diff
@@ -170,12 +170,12 @@ async function getBookings({
+    ? []
+    : Object.keys(filters)
+        .map((key) => bookingWhereInputFilters[key])
+        // On prisma 5.4.2 passing undefined to where "AND" causes an error
```
Left a note here

PeerRich committed:
Merge branch 'main' into 10-02-chore_upgrade_to_prisma_5.3.1

ThyMinimalDev reviewed (approved):
Looks good to me 👍 

checked out the branch locally and tested ✅
